frontend/micro-ui/web/micro-ui-internals/packages/modules/hearings/src/pages/employee/InsideHearingMainPage.js

```diff
@@ -207,14 +207,14 @@ const InsideHearingMainPage = () => {
   }, [transcriptText, setTranscriptText]);
 
   const isDepositionSaved = useMemo(() => {
-    return hearing?.additionalDetails?.witnessDepositions?.find((witness) => witness.uuid === selectedWitness.uuid)?.deposition.length;
+    return Boolean(hearing?.additionalDetails?.witnessDepositions?.find((witness) => witness.uuid === selectedWitness.uuid)?.deposition);
   }, [selectedWitness, hearing]);
 
   const saveWitnessDeposition = () => {
-    const updatedHearing = structuredClone(hearing);
+    const updatedHearing = structuredClone(hearing || {});
```

Comment:
why are we saving witness deposition if hearing is null/undefined ?

it should be the first check in this function
```js
if (!hearing) return
```

Reply:
addressed